lxd/project/limits/permissions.go

```diff
@@ -1233,10 +1241,11 @@ func fetchProject(globalConfig map[string]any, tx *db.ClusterTx, projectName str
 	}
 
 	info := &projectInfo{
-		Project:   *project,
-		Profiles:  profiles,
-		Instances: instances,
-		Volumes:   volumes,
+		Project:      *project,
+		Profiles:     profiles,
+		StoragePools: pools,
```

Comment:
Should this be limited to only pools which are actually used in the project? 
I guess `tx.GetStoragePools` returns all of them.
Not sure if there is a filter option like for the profiles.

Reply:
I don't see a function under `lxd/db` that would do that. I took a look through the database schema and [the docs](https://documentation.ubuntu.com/lxd/en/latest/reference/projects/#project-features) and there isn't a clear/direct relationship between projects and pools; this would likely mean going through all the devices in the project to find the pools in use.

I can implement a filter if you'd like but I don't think the complexity is worth it.

Reply:
My thinking was if such a `projectInfo` struct gets initialized it's probably used with the intention that everything in there is part of the project. Same as for profiles, volumes and instances.
But as storage pools are shared across projects that kind of breaks this logic.

Maybe you can do the DB lookup in `getInstanceLimits` directly?

Reply:
I'll give that a shot; it also saves us from doing an extra query that we only need in a special case.

Reply:
We dont want to be doing any extra DB queries in `getInstanceLimits` as its run per instance in the project.

I dont see an issue with storing a pool name -> driver name map on `projectInfo` even if it does have pools in it that are not used by the project (a comment could help here).

Then we can add a specialised function that retrieves just the needed info in a single query.